add version 1 of dynamic rule (#2334)
port-labs/port-docs: docs/build-your-software-catalog/set-catalog-rbac/set-catalog-rbac.md

:::

To give a <b>dynamic</b> `read` permissions to users, change the `policy` key - with the `policy` key you can add search query as dynamic permission policy.
:::tip
you can use [contextual query rules](/search-and-query/#contextual-query-rules), to have the context of the user executing the query.
:::

in this example `on-call` users are grented `read` access only to entities that share the same `region` as one of their owning teams:

```json showLineNumbers
{
  "entities": {
    ... other permissions
    "read": {
      "roles": ["my-blueprint-moderator", "Admin"],
      "users": [],
      "teams": [],
      "ownedByTeam": false,
      // highlight-start
      "policy": {
        "combinator": "and",
        "rules": [
          {
            "property": {
              "context" : "user",
              "property": "isOnCall"
            },
            "operator": "=",
            "value": "true"
          },
          {
            "property": "region",
            "operator": "containsAny",
            "value": {
              "context" : "userteams",
              "property": "region"
            }
          },
        ]
      }
      // highlight-end
    }
  }
}
```

> :::info Policy taking effect
> policy will take effect only if the user is not effected by any other `read` permission key.
> if the user is having full `read` permisison or `ownedByTeam` permission this query will not take effect.
> :::

Reviewer: ```suggestion
:::info Policy usage
The `policy` key will take effect only if the user is not affected by any other `read` permission keys.  
For example, if the user has full `read` permissions or `ownedByTeam` permissions, this query will not take effect.
:::
```
